pkg/types/baremetal/validation/platform.go

```diff
@@ -32,6 +33,26 @@ func validateIPNotinMachineCIDR(ip string, n *types.Networking) error {
 	return nil
 }
 
+func validateOSImageURI(uri string) error {
+	// Check for valid URI and sha256 checksum part of the URL
+	var sha256Checksum string
+	parsedURL, err := url.ParseRequestURI(uri)
+
+	if err != nil {
+		return fmt.Errorf("The URI provided is invalid")
+	}
+
+	if sha256Checksums, ok := parsedURL.Query()["sha256"]; ok {
+		sha256Checksum = sha256Checksums[0]
+	}
+
+	if sha256Checksum == "" || len(sha256Checksum) != 64 {
+		return fmt.Errorf("The sha256 parameter in the URI is either invalid or missing")
+	}
```

Comment:
would like to see 2 different messages, not found and invalid.

Reply:
Will do.